pkg/operation/botanist/component/etcdcopybackupstask/etcdcopybackupstask.go

```diff
@@ -0,0 +1,207 @@
+	SourceStore druidv1alpha1.StoreSpec
+	// TargetStore is the specification of the objcet store to which etcd backups will be copied
+	TargetStore druidv1alpha1.StoreSpec
+	// MaxBackups is the maximum number of backups that will be copied starting with the most recent ones.
+	MaxBackups *uint32
+	// MaxBackupAge is the maximum age in days that a backup must have in order to be copied.
+	MaxBackupAge *uint32
+}
+
+type etcdCopyBackupsTask struct {
+	values       *Values
+	logger       logrus.FieldLogger
+	client       client.Client
+	waitInterval time.Duration
+	waitTimeout  time.Duration
+
+	task *druidv1alpha1.EtcdCopyBackupsTask
+}
+
+// New creates a new instance of Interface
+func New(
+	logger logrus.FieldLogger,
+	client client.Client,
+	values *Values,
+	waitInterval time.Duration,
+	waitTimeout time.Duration) Interface {
+
+	return &etcdCopyBackupsTask{
+		values,
+		logger,
+		client,
+		waitInterval,
+		waitTimeout,
+		&druidv1alpha1.EtcdCopyBackupsTask{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      values.Name,
+				Namespace: values.Namespace,
+			},
+		},
+	}
+}
+
+// Deploy creates the EtcdBackupsCopyTask resource. If it already exists, it is deleted and then recreated.
+func (e *etcdCopyBackupsTask) Deploy(ctx context.Context) error {
+	if err := e.Destroy(ctx); err != nil {
+		return err
+	}
+
+	e.task.Spec.MaxBackupAge = e.values.MaxBackupAge
+	e.task.Spec.MaxBackups = e.values.MaxBackups
+	e.task.Spec.SourceStore = e.values.SourceStore
+	e.task.Spec.TargetStore = e.values.TargetStore
+	return e.client.Create(ctx, e.task)
+}
+
+// Wait waits until the EtcdBackupsCopyTask is ready
+func (e *etcdCopyBackupsTask) Wait(ctx context.Context) error {
+	if err := extensions.WaitUntilObjectReadyWithHealthFunction(
+		ctx,
+		e.client,
+		e.logger,
+		WaitForConditions,
+		e.task,
+		"EtcdCopyBackupsTask",
+		e.waitInterval,
+		e.waitTimeout,
+		e.waitTimeout,
+		e.CheckConditions,
+	); err != nil {
+		e.logger.Error(err)
+		return err
+	}
+	return nil
+}
+
+// Destroy deletes the EtcdBackupsCopyTask resource.
+func (e *etcdCopyBackupsTask) Destroy(ctx context.Context) error {
+	return kutil.DeleteObject(
+		ctx,
+		e.client,
+		e.task,
+	)
+}
+
+// WaitCleanup is not implemeted yet.
+func (e *etcdCopyBackupsTask) WaitCleanup(_ context.Context) error { return nil }
+
+// SetSourceStore sets the specifications for the object store provider from which backups will be copied
+func (e *etcdCopyBackupsTask) SetSourceStore(store druidv1alpha1.StoreSpec) {
+	e.values.SourceStore = store
+}
+
+// SetTargetStore sets the specifications for the object store provider to which backups will be copied
+func (e *etcdCopyBackupsTask) SetTargetStore(store druidv1alpha1.StoreSpec) {
+	e.values.TargetStore = store
+}
+
+// WaitForConditions waits until the EtcdCopyBackupsTask conditions have been populated by the etcd druid
+func WaitForConditions(obj client.Object) error {
+	etcdCopyBackupsTask, ok := obj.(*druidv1alpha1.EtcdCopyBackupsTask)
```

Comment:
Let's call this `task`, it's shorter and avoids the confusion with the receiver names of other methods (which however have a different type)
```suggestion
	task, ok := obj.(*druidv1alpha1.EtcdCopyBackupsTask)
```